contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,952 @@
+        address _oethbAddress,
+        address _swapRouter,
+        address _nonfungiblePositionManager,
+        address _clPool,
+        address _clGauge,
+        address _sugarHelper,
+        int24 _lowerBoundingTick,
+        int24 _upperBoundingTick,
+        int24 _tickClosestToParity
+    ) InitializableAbstractStrategy(_stratConfig) {
+        WETH = _wethAddress;
+        OETHb = _oethbAddress;
+        swapRouter = ISwapRouter(_swapRouter);
+        positionManager = INonfungiblePositionManager(
+            _nonfungiblePositionManager
+        );
+        clPool = ICLPool(_clPool);
+        clGauge = ICLGauge(_clGauge);
+        helper = ISugarHelper(_sugarHelper);
+        sqrtRatioX96Tick0 = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(
+            _lowerBoundingTick
+        );
+        sqrtRatioX96Tick1 = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(
+            _upperBoundingTick
+        );
+        sqrtRatioX96TickClosestToParity = ISugarHelper(_sugarHelper)
+            .getSqrtRatioAtTick(_tickClosestToParity);
+
+        lowerTick = _lowerBoundingTick;
+        upperTick = _upperBoundingTick;
+        tickSpacing = 1;
+
+        require(
+            ICLPool(_clPool).token0() == _wethAddress,
+            "Only WETH supported as token0"
+        );
+        require(
+            ICLPool(_clPool).token1() == _oethbAddress,
+            "Only OETHb supported as token1"
+        );
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     * @param _rewardTokenAddresses Address of reward token for platform
+     */
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory,
+        address[] memory
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            // these should all be empty
+            new address[](0),
+            new address[](0)
+        );
+    }
+
+    /***************************************
+                  Configuration 
+    ****************************************/
+
+    /**
+     * @notice Set the new desired WETH share
+     * @param _amount The new amount specified in basis points
+     */
+    function setPoolWethShare(uint256 _amount) external onlyGovernor {
+        require(_amount < 1e18, "Invalid poolWethShare amount");
+        require(_amount > 0, "Invalid poolWethShare amount");
+
+        poolWethShare = _amount;
+        emit PoolWethShareUpdated(_amount);
+    }
+
+    /**
+     * @notice Specifies the amount of liquidity that is to be removed when
+     *         a rebalancing happens.
+     * @param _amount The new amount specified in basis points
+     */
+    function setWithdrawLiquidityShare(uint128 _amount) external onlyGovernor {
+        require(_amount < 1e18, "Invalid withdrawLiquidityShare amount");
+
+        withdrawLiquidityShare = _amount;
+        emit WithdrawLiqiudityShareUpdated(_amount);
```

Comment:
🟠 This seems like `withdrawLiquidityShare` has no reason to be a variable. It could very well be a param to `rebalance` method.

Could remove this setter method as well that way

Reply:
Yeah I agree and it can be set by the strategist, the same account that does the rebalancing. 

Though I do like the `withdrawLiquidityShare`  being set explicitly set by another function. I feel like the explicitness makes it slightly less error prone? What do you think?

Reply:
If the strategist has to set it every time a rebalance happens, I still don't see how it'd be less error prone though

Reply:
The current plan is for the strategist to set it less often than it rebalances. It will probably set it to a lower value initially - smth like 5%WETH 95% OETHb. And then after a couple of rounds to something closer to 20% WETH and 80% OETHb and keep it there. And there should be multiple rebalances on those token ratios.
That is at least how we are thinking about it right now.

Reply:
I like that idea. We do have such a token for the pool now: https://basescan.org/token/0x827922686190790b37229fd06084350e74485b72?a=413296#inventory owned by a dead address. 

Will add appropriate documentation to the contract that such liquidity in the trading ticker is part of setup requirements. And remove the complexity in the code where not whole LP position is removed.

Reply:
I've made the change in a separate PR since it is more involved: https://github.com/OriginProtocol/origin-dollar/pull/2198. @shahthepro & @DanielVF would you mind reviewing the changes and approving the linked PR?